app/services/cart_items_management_service.rb

```diff
@@ -1,30 +1,39 @@
 class CartItemsManagementService
-  attr_reader :cart, :product, :quantity
+  attr_reader :cart, :product, :quantity, :cart_product
 
-  def initialize(cart, product, quantity)
+  def initialize(cart, product, quantity = 0)
     @cart = cart
     @product = product
     @quantity = quantity
+    @cart_product = CartProduct.find_by(cart: cart, product: product)
+  end
 
-    validate_quantity!
+  def create_product_in_cart
```

Comment:
regarding the name of the service is giving us the context that we are handling the cart contents, i would keep the names a bit shorter. wdyt?
```suggestion
  def create_product!
```
also, add the bang too

Reply:
updated.